### PR TITLE
Fix cxx11 abi

### LINKED ISSUE #55829: [nightlies] builds for libtorch cxx11-abi failing (Sort.cu)
## 🐛 Bug

Builds are currently failing with this ([CircleCI Logs](https://app.circleci.com/pipelines/github/pytorch/pytorch/299564/workflows/387663b3-e0d3-4071-a290-ea9fd543c7d5/jobs/12299829)):
```
FAILED: caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/ATen/native/cuda/torch_cuda_generated_Sort.cu.o 
cd /pytorch/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/ATen/native/cuda && /opt/conda/bin/cmake -E make_directory /pytorch/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/ATen/native/cuda/. && /opt/conda/bin/cmake -D verbose:BOOL=OFF -D build_configuration:STRING=Release -D generated_file:STRING=/pytorch/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/ATen/native/cuda/./torch_cuda_generated_Sort.cu.o -D generated_cubin_file:STRING=/pytorch/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/ATen/native/cuda/./torch_cuda_generated_Sort.cu.o.cubin.txt -P /pytorch/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/ATen/native/cuda/torch_cuda_generated_Sort.cu.o.Release.cmake
/pytorch/aten/src/ATen/native/cuda/Sort.cu(102): error: copy-list-initialization cannot use a constructor marked "explicit"

/pytorch/aten/src/ATen/native/cuda/Sort.cu(156): error: copy-list-initialization cannot use a constructor marked "explicit"

/pytorch/aten/src/ATen/native/cuda/Sort.cu(209): error: copy-list-initialization cannot use a constructor marked "explicit"

3 errors detected in the compilation of "/tmp/tmpxft_000031a8_00000000-9_Sort.compute_70.cpp1.ii".
CMake Error at torch_cuda_generated_Sort.cu.o.Release.cmake:281 (message):
  Error generating file
  /pytorch/build/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/ATen/native/cuda/./torch_cuda_generated_Sort.cu.o
```

Build failure was introduced in https://github.com/pytorch/pytorch/pull/54626

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @anjali411 @ngimel @bhosmer @smessmer @ljk53 @ailzhang @zasdfgbnm

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Fixes https://github.com/pytorch/pytorch/issues/55829

### CODE REVIEW

**Review by ngimel — APPROVED** [review]
lgtm, @seemethere can you check if it fixes nightly builds?

**Review by seemethere — APPROVED** [review]

**Review by malfet — CHANGES_REQUESTED** [review]
Hmm, perhaps rather than fixing syntactic error, function signature better be fixed?
Returning reference to object from function is a bad practice, that can lead to hard to catch errors. Which makes me wonder, why it returns those in the first place.

**Comment by zasdfgbnm** [comment]
@malfet Isn't returning reference of tensors the required behavior for all `xxx_out` operators?

```python
>>> import torch
>>> a = torch.randn(5, 5)
>>> b = torch.randn(5)
>>> torch.sum(a, dim=0, out=b).requires_grad_()
tensor([-2.3231e-01, -2.1876e+00,  1.7415e+00, -3.2706e-03,  3.7022e+00],
       requires_grad=True)
>>> b
tensor([-2.3231e-01, -2.1876e+00,  1.7415e+00, -3.2706e-03,  3.7022e+00],
       requires_grad=True)
```

File: aten/src/ATen/native/cuda/Sort.cu [hunk 1]
```diff
@@ -99,7 +99,7 @@ std::tuple<Tensor &,Tensor &> sort_out_stable_cuda(const Tensor & self, c10::opt
       indices.resize_as_(self);
       indices.zero_();
     }
-    return {values, indices};
+    return std::tuple<Tensor &, Tensor &>(values, indices);
```
Comment (malfet) [inline aten/src/ATen/native/cuda/Sort.cu:102]: Please use [std::forward_as_tuple](https://en.cppreference.com/w/cpp/utility/tuple/forward_as_tuple)
```suggestion
    return std::forward_as_tuple(values, indices);
```

File: aten/src/ATen/native/cuda/Sort.cu [hunk 2]
```diff
@@ -153,7 +153,7 @@ std::tuple<Tensor &,Tensor &> sort_out_stable_cuda(const Tensor & self, c10::opt
   }
 
   if (numel == 0) {
-    return {values, indices};
+    return std::tuple<Tensor &, Tensor &>(values, indices);
```
Comment (malfet) [inline aten/src/ATen/native/cuda/Sort.cu:156]: Same as above
```suggestion
    return std::forward_as_tuple(values, indices);
```

File: aten/src/ATen/native/cuda/Sort.cu [hunk 3]
```diff
@@ -206,7 +206,7 @@ std::tuple<Tensor &,Tensor &> sort_out_stable_cuda(const Tensor & self, c10::opt
   if (indices_tmp.defined()) {
     indices.copy_(indices_tmp);
   }
-  return {values, indices};
+  return std::tuple<Tensor &, Tensor &>(values, indices);
```
Comment (malfet) [inline aten/src/ATen/native/cuda/Sort.cu:209]: Same as above
```suggestion
  return std::forward_as_tuple(values, indices);
```

**Review by malfet — APPROVED** [review]
LGTM, but please use `std::forward_as_tuple` to make sure those are returned as references

**Comment by zasdfgbnm** [comment]
@malfet fixed

**Comment by seemethere** [comment]
Looks like there might be a `clang-format` issue?

**Pull request closed without merging**